pkg/common/constants.go

```diff
@@ -49,3 +49,8 @@ const (
 
 var SessionDataKey contextKey
 var SessionInternalKey contextKey
+
+//Regex
+const (
+	PROJECT_NAME_REGEX = `^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`
```

Comment:
this regex looks good, except that name like `my-$-project` is allowed which inturn gets replaced and created with the name my-himBHsproject which we should not be allowing.

let the valid characters be same as given regex and hyphen, except at the beginning or the end. No other special chars allowed.

Reply:
@hiteshwani29 Any reason why all letters except first are grouped and added `()?` quantifier? Can this just be `^[a-zA-Z][-a-zA-Z0-9]*[a-zA-Z0-9]$`

Reply:
No @akshay196 not because of any specific reason I will make the changes in regex respectively .

Reply:
@niravparikh05 
Not sure but the cobra library is not parsing string arguments correctly with special characters such as '$' or '$$'. 
- for `pctl creat project my-$-project ` It replaces the $ with some string char and gives us some incorrect name in args array.
- but if we pass the project name with escape char `my-\$-project` or with a single quote `'my-$-project'` in the command. It parses the arg correctly in the args array and throws a validating error